ratelimit/token_bucket.go

```diff
@@ -18,25 +18,100 @@ var ErrLimited = errors.New("rate limit exceeded")
 // limiter based on a token-bucket algorithm. Requests that would exceed the
 // maximum request rate are simply rejected with an error.
 func NewTokenBucketLimiter(tb *ratelimit.Bucket) endpoint.Middleware {
+	return NewErroringLimiter(NewAllower(tb))
+}
+
+// NewTokenBucketThrottler returns an endpoint.Middleware that acts as a
+// request throttler based on a token-bucket algorithm. Requests that would
+// exceed the maximum request rate are delayed via the parameterized sleep
+// function. By default you may pass time.Sleep.
+func NewTokenBucketThrottler(tb *ratelimit.Bucket, sleep func(time.Duration)) endpoint.Middleware {
+	// return NewDelayingLimiter(NewWaiter(tb))
```

Comment:
I can do it, but then the `sleep` parameter gets ignored. I'll check in that change, so you can evaluate if you like it.